### PR TITLE
Log indexes test

### PR DESCRIPTION
**Pull request by kjazgar**
Fixes | Closes | Resolves #

## Changes:
Changed enumerating of log indexes. Added test for ReceiptsForRpc class checking if log indexes are enumerated properly.

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe)

## Types of changes

Requires testing

- [x] Yes
- [ ] No

In case you checked yes, did you write tests??

- [x] Yes
- [ ] No

### CODE REVIEW

**Review by LukaszRozmej — CHANGES_REQUESTED** [review]
We need one with multiple receipts per block

**Comment by MarekM25** [comment]
> We need one with multiple receipts per block

Right !

**Review by MarekM25 — CHANGES_REQUESTED** [review]
@LukaszRozmej has the right

File: src/Nethermind/Nethermind.JsonRpc/Modules/Proof/ProofRpcModule.cs [hunk 1]
```diff
@@ -143,6 +144,23 @@ public ResultWrapper<TransactionWithProof> proof_getTransactionByHash(Keccak txH
 
             return ResultWrapper<TransactionWithProof>.Success(txWithProof);
         }
+        
+        private int SumOfPreviousLogIndexes(Block block, TxReceipt txReceipt)
```
Comment (MarekM25) [inline src/Nethermind/Nethermind.JsonRpc/Modules/Proof/ProofRpcModule.cs:148]: Can we share this logic across all modules?

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.JsonRpc/Modules/Proof/ProofRpcModule.cs:148]: Share the logic between the parity one too

File: src/Nethermind/Nethermind.Facade/BlockchainBridge.cs [hunk 2]
```diff
@@ -118,21 +118,38 @@ public Block BeamHead
...
+                {
+                    sum += receipts[i].Logs.Length;
+                }
+            }
+
+            return sum;
+        }
 
-        public (TxReceipt Receipt, UInt256? EffectiveGasPrice) GetReceiptAndEffectiveGasPrice(Keccak txHash)
+        public (TxReceipt Receipt, UInt256? EffectiveGasPrice, int SumOfPreviousLogIndexes) GetReceiptAndEffectiveGasPrice(Keccak txHash)
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Facade/BlockchainBridge.cs:137]: naming: (First?)LogIndexInBlock?

**Review by LukaszRozmej — APPROVED** [review]

**Pull request merged**